### PR TITLE
Update to allow different merge methods in pulls_service

### PR DESCRIPTION
**Pull request by ricardoamador**
Updating the merge pull request service to allow the merge method to be specified when merging a pull request.
Also allowing the sha to be specified and adding the recommended headers (by github) when making the request to the service.

### CODE REVIEW

File: lib/src/common/pulls_service.dart [hunk 1]
```diff
@@ -184,3 +196,9 @@ class PullRequestsService extends Service {
     );
   }
 }
+
+enum MergeMethod {
```
Comment (CaseyHillers) [inline lib/src/common/pulls_service.dart:200]: Can you add a dartdoc with a link to the GitHub reference

File: lib/src/common/pulls_service.dart [hunk 2]
```diff
@@ -121,15 +121,27 @@ class PullRequestsService extends Service {
...
     }
+    if (requestSha != null) {
+      json['sha'] = requestSha;
+    }
+
+    json['merge_method'] = mergeMethod.name;
+
+    // Recommended Accept header when making a merge request.
+    Map<String, String>? headers = <String, String>{};
+    headers['Accept'] = 'application/vnd.github+json';
```
Comment (CaseyHillers) [inline lib/src/common/pulls_service.dart:140]: Can we add a test?

File: lib/src/common/pulls_service.dart (hunk 1 above)
Comment (ricardoamador) [reply lib/src/common/pulls_service.dart:200]: Yeah I can do that.

File: lib/src/common/pulls_service.dart (hunk 2 above)
Comment (ricardoamador) [reply lib/src/common/pulls_service.dart:140]: I should be able to.

**Review by robrbecker — APPROVED** [review]
pulls_test passes

**Comment by robrbecker** [comment]
Sorry I didn't get to this sooner. Are you happy with it too @CaseyHillers? Ready for me to merge?

**Comment by robrbecker** [comment]
FYI It LGTM so if I don't hear from you today, I'm just going to merge and release it.

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**